### ISSUE TITLE
[HOLD for payment 2024-07-17] [$250] Profile - Profile changes to Hidden and "Email not found" error when messaging Hidden user

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43278]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.80-9
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to FAB > Start chat
3. Enter a non-existing email
4. Select the email
5. Quickly click on the chat header after landing on the DM
6. After the profile changes to Hidden, click Message

## Expected Result:
In Step 5, the user profile RHP will remain unchanged

## Actual Result:
In Step 5, the user profile RHP changes to Hidden
When clicking Message, app redirects to chat with another user with the error "Email not found"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/5ead81bf-ce1f-4084-a661-462d2db246ca

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018004b4dcf5d57a3b</li>
        <li>Upwork Job ID: 1800107416738330577</li>
        <li>Last Price Increase: 2024-07-01</li>
<li>Automatic offers: </li>
<ul>
<li>ZhenjaHorbach | Reviewer | 102957514</li>
<li>bernhardoj | Contributor | 102957518</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @twisterdotcom</details>

**Comment by lanitochka17** [comment]
@laurenreidexpensify   FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Profile page changes to hidden and error when pressing Message on a new user.

### What is the root cause of that problem?
When we start a new chat, we will create an optimistic personal detail for the user. We actually have a disabled condition to disable the press when the user is optimistic.
https://github.com/Expensify/App/blob/bb6e7eb5d75689d1b01fd1dcd31e657750a92c79/src/pages/home/HeaderView.tsx#L258
https://github.com/Expensify/App/blob/bb6e7eb5d75689d1b01fd1dcd31e657750a92c79/src/libs/ReportUtils.ts#L1124-L1132

However, the `participants` contains both the user and the current user, so the length is 0, thus the `participantAccountIDs.length === 1` condition is never satisfied. This happens after we migrate from `report.participantAccountIDs` to `report.participants`.

### What changes do you think we should make in order to solve the problem?

Filter out current user from `report.participants`. We can also optionally replace `participantAccountIDs.length === 1` with `isOneOnOneChat`.

```
if (isOneOnOneChat(report)) {
    const participantAccountIDs = Object.keys(report?.participants ?? {}).map(Number).filter((accountID) => accountID !== currentUserAccountID);
    return isOptimisticPersonalDetail(participantAccountIDs[0]);
}
```

**Comment by bernhardoj** [comment]
Yeah, maybe it's not the best behavior, but just want to point out that it's how it works before (first introduced in https://github.com/Expensify/App/issues/22691)

**Comment by laurenreidexpensify** [comment]
@Expensify/design curious for your take on this one wrt to the flow here and @ZhenjaHorbach's comment: 

> Because after we open the chat we will have a non-clickable header and a user will not understand why it does not work
And this will last until the user waits for a response from the server

Do you agree?

**Comment by dannymcclain** [comment]
Hmm, I'm not sure I totally understand what's going on. Why does the user change from an email address to "Hidden"?

**Comment by shawnborton** [comment]
Yeah, I don't quite understand that part either. @puneetlath I think you worked on some of the "Hidden" stuff previously, do you have an opinion here?

**Comment by bernhardoj** [comment]
> Why does the user change from an email address to "Hidden"?

To add more information, when we start a chat with a new user, we optimistically create the user information on the FE (accountID, etc.). The BE then responds with different information and we clear the optimistic information to avoid keeping unused information.

So for example, the optimistic accountID is 1, but the real accountID sent from the BE is 2.

**Comment by twisterdotcom** [comment]
Okay, is this waiting on me, or @puneetlath or somebody else? I'm confused, Lauren was the BZ but wasn't answering the earlier Q's from @shawnborton or @ZhenjaHorbach.

**Comment by twisterdotcom** [comment]
Okay, is this waiting on me, or @puneetlath or somebody else? I'm confused, Lauren was the BZ but wasn't answering the earlier Q's from @shawnborton or @ZhenjaHorbach.

**Comment by twisterdotcom** [comment]
Okay, sounds good. Let's revert, unless @puneetlath intervenes in the meantime.

**Comment by arosiclair** [comment]
[Proposal](https://github.com/Expensify/App/issues/43278#issuecomment-2154932906) makes sense to me. @marcaaron I think you worked on the participants/participantAccountIDs refactor. You mind taking a look before I hire?

**Comment by marcaaron** [comment]
It also makes sense to me. 

Although, I am not sure if we have settled on exactly how this behavior should work. It recently came up in another issue that @techievivek was leading.

In particular, @shawnborton asked why we do not allow navigating to the profile view in [this comment](https://github.com/Expensify/App/issues/38925#issuecomment-2196247877) (that was in regards to an "optimistic" user - but perhaps the same logic should apply to a "Hidden" user - whatever that means).

I don't super passionate about this and would suggest we continue to block navigating to profiles for "optimistic" users as it's the simplest solution and requires the least amount of accommodation, but I'm just one voice in this convo.

**Comment by arosiclair** [comment]
> I don't super passionate about this and would suggest we continue to block navigating to profiles for "optimistic" users as it's the simplest solution and requires the least amount of accommodation, but I'm just one voice in this convo.

Yeah that seems best at least for now. Hiring @bernhardoj

**Comment by bernhardoj** [comment]
PR is ready

cc: @ZhenjaHorbach

**Comment by bernhardoj** [comment]
I'll request in ND once payment is due.

**Comment by bernhardoj** [comment]
The payment should be due [tomorrow](https://github.com/Expensify/App/issues/43278#issuecomment-2220196939) (07-17)

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by twisterdotcom** [comment]
Payment Summary:

- @bernhardoj paid $250 via NewDot Manual Request (Contributor)
- @ZhenjaHorbach paid $250 [here](https://www.upwork.com/nx/wm/offer/102957514) (Reviewer)

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
